source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1237 @@
+          TableIndices<dim> size_idx;
+          for (unsigned int d=0; d<dim; ++d)
+            {
+              size_idx[d] = table_intervals[d]+1;
+            }
+
+          // Initialize a table to hold all velocity values that will be interpolated back to ASPECT.
+          Table<dim,double> data_table;
+          data_table.TableBase<dim,double>::reinit(size_idx);
+          TableIndices<dim> idx;
+
+          // Loop through the data table and fill out the surface (i or k = 1) to the velocities from FastScape.
+          if (dim == 2)
+            {
+              std::vector<double> V2(nx);
+
+              for (int i=1; i<(nx-1); i++)
+                {
+                  // If using the center slice, find velocities from the row closest to the center.
+                  if (slice)
+                    {
+                      const int index = i+nx*(round((ny-1)/2));
+                      V2[i-1] = V[index];
+                    }
+                  // Here we use average velocities across the y nodes, excluding the ghost nodes (top and bottom row).
+                  // Note: If ghost nodes are turned off, boundary effects may influence this.
+                  else
+                    {
+                      for (int ys=1; ys<(ny-1); ys++)
+                        {
+                          const int index = i+nx*ys;
+                          V2[i-1] += V[index];
+                        }
+                      V2[i-1] = V2[i-1]/(ny-2);
+                    }
+                }
+
+              for (unsigned int i=0; i<data_table.size()[1]; ++i)
+                {
+                  idx[1] = i;
+
+                  for (unsigned int j=0; j<(data_table.size()[0]); ++j)
+                    {
+                      idx[0] = j;
+
+                      // Convert from m/yr to m/s if needed. i == 1 is the surface and i = 0 the bottom.
+                      if (i == 1)
+                        {
+                          if (this->convert_output_to_years())
+                            data_table(idx) = V2[j]/year_in_seconds;
+                          else
+                            data_table(idx) = V2[j];
+                        }
+                      else
+                        data_table(idx)= 0;
+                    }
+                }
+            }
+          else
+            {
+              // Indexes through z, y, and then x.
+              for (unsigned int k=0; k<data_table.size()[2]; ++k)
+                {
+                  idx[2] = k;
+
+                  for (unsigned int i=0; i<data_table.size()[1]; ++i)
+                    {
+                      idx[1] = i;
+
+                      for (unsigned int j=0; j<data_table.size()[0]; ++j)
+                        {
+                          idx[0] = j;
+
+                          // Tables are initialized with k = 0 as the bottom and 1 as top, we are only interested in the surface so only fill k==1.
```

Comment:
Do you ever need the bottom? Can you make a table without it?

Reply:
I may be overlooking something, but the interpolate_boundary_values always causes a crash if I try to only have a top. Otherwise the bottom shouldn't really be needed.